File: Front-end/angularshop/src/app/employeesschedule.ts
import { Time } from "@angular/common";

export class EmployeesSchedule {
    
    id:number;
    employee_id:number;
    day_id:number;
    from_hour:Time;
    to_hour:Time;
    constructor(){
        this.day_id=0;
        this.id=0;
        this.employee_id=0;
        this.from_hour = { hours: 0, minutes: 0}; // Initialize with default time
        this.to_hour = { hours: 0, minutes: 0 }; // Initialize with default time
    }
}

File: Back-end/user_service/src/main/java/com/springboot/entities/Employees_schedule.java
package com.springboot.entities;

import java.sql.Date;
import java.sql.Time;
import java.sql.Timestamp;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "employees_schedule")
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Employees_schedule {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;


	@ManyToOne
    @JoinColumn( name="employee_id", nullable=false )
	private Employee employee;

	private Time from_hour;
	private Time to_hour;
	private int day_id;
	
	

}
